sympy/solvers/simplex.py

```diff
@@ -0,0 +1,966 @@
+            Inconsistent/False constraint"""))
+    np, r, aux = _
+
+    # do change of variables
+    F = F.xreplace(r)
+    np = [i.xreplace(r) for i in np]
+
+    # convert to matrices
+    xx = list(ordered(syms)) + aux
+    A, B = linear_eq_to_matrix(np, xx)
+    C, D = linear_eq_to_matrix([F], xx)
+    return A, B, C, D, r, xx, aux
+
+
+def _lp(min_max, f, constr):
+    """Return the optimization (min or max) of ``f`` with the given
+    constraints. Unless constrained, variables will assume only
+    nonnegative values in the solution. A constraint like ``x <= 2``
+    will permit ``x`` to have values in range ``[-oo, 2]`` while
+    ``x < oo`` will allow it to have any real value.
+
+    If `min_max` is 'max' then the results corresponding to the
+    maximization of ``f`` will be returned, else the minimization.
+    The constraints can be given as Le, Ge or Eq expressions.
+
+    Examples
+    ========
+
+    >>> from sympy.solvers.simplex import _lp as lp
+    >>> from sympy import Eq
+    >>> from sympy.abc import x, y, z
+    >>> f = x + y - 2*z
+    >>> c = [7*x + 4*y - 7*z <= 3, 3*x - y + 10*z <= 6]
+    >>> lp(min, f, c)
+    (-6/5, {x: 0, y: 0, z: 3/5})
+
+    By passing max, the maximum value for f under the constraints
+    is returned (if possible):
+
+    >>> lp(max, f, c)
+    (3/4, {x: 0, y: 3/4, z: 0})
+
+    Constraints that are equalities will require that the solution
+    also satisfy them:
+
+    >>> lp(max, f, c + [Eq(y - 9*x, 1)])
+    (5/7, {x: 0, y: 1, z: 1/7})
+
+    All symbols are reported, even if they are not in the objective
+    function:
+
+    >>> lp(min, x, [y + x >= 3])
+    (0, {x: 0, y: 3})
+    """
+    A, B, C, D, r, xx, aux = _lp_matrices(f, constr)
+
+    how = str(min_max).lower()
+    if 'max' in how:
+        # _simplex minimizes for Ax <= B so we
+        # have to change the sign of the function
+        # and negate the optimal value returned
+        _o, p, d = _simplex(A, B, -C, -D)
+        o = -_o
+    elif 'min' in how:
+        o, p, d = _simplex(A, B, C, D)
+    else:
+        raise ValueError('expecting min or max')
+
+    # restore original variables and remove aux from p
+    p = dict(zip(xx, p))
+    if r:  # p has original symbols and auxilliary symbols
+        # if r has x: x - z1 use values from p to update
+        r = {k: v.xreplace(p) for k, v in r.items()}
+        # then use the actual value of x (= x - z1) in p
+        p.update(r)
+        # don't show aux
+        p = {k: p[k] for k in ordered(p) if k not in aux}
+
+    # not returning dual since there may be extra constraints
+    # when a variable has finite bounds
+    return o, p
+
+
+def lpmin(f, constr):
+    """return minimum of linear equation ``f`` under
+    linear constraints expressed using Ge, Le or Eq.
+
+    All variables will take on nonnegative values unless
+    a univariate constraint indicates other ranges for
+    a symbol, e.g, ``x <= 3`` will allow for negative
```

Comment:
Is it simple to avoid assuming nonnegativity by default?

Reply:
It is not hard to do so. It is harder to think what will make the most sense to people using the function. The inequality `x <= 3` clearly permits negative values. So I would rather that the user also indicate x >= 0 if they also want that condition. If I automatically add `x >= 0` then how will the user indicate a range like -1 to 1?

Reply:
> It is harder to think what will make the most sense to people using the function

I think most users would assume that since they are providing assumptions the function will use the provided assumptions rather than imposing any assumptions a priori.

Reply:
Re-reading this, I think I miscued on your intent. I think you are saying that the variables should be unbounded unless constrained otherwise. This is as simple as adding `xi <= oo` for all `xi`. If they want nonnegative values then they should explicitly say `xi >= 0`.

Reply:
I would expect the method not to assume anything other than that symbols are real by default. It shouldn't be necessary to pass `xi <= oo` in order to specify that `xi` might be negative.

Reply:
This, now, is how the routine behaves as mentioned in my comment (way below). I totally agree.